src/SkeletalTrapezoidation.cpp

```diff
@@ -393,10 +510,71 @@ void SkeletalTrapezoidation::constructFromPolygons(const Polygons& polys)
         }
     }
 
-    vd_t vonoroi_diagram;
-    construct_voronoi(segments.begin(), segments.end(), &vonoroi_diagram);
+    vd_t voronoi_diagram;
+    construct_voronoi(segments.begin(), segments.end(), &voronoi_diagram);
 
-    for (vd_t::cell_type cell : vonoroi_diagram.cells())
+    // Try to detect cases when some Voronoi vertex is missing.
+    // When any Voronoi vertex is missing, rotate input polygon and try again.
+    constexpr double fix_angle = 0.5;  // Choose an off-kilter (non-Pi/non-Tau based) angle to counteract rotational symmetry.
+    const bool has_missing_voronoi_vertex = detectMissingVoronoiVertex(voronoi_diagram, points, segments);
+    const bool is_voronoi_planar = isVoronoiDiagramPlanarAngle(voronoi_diagram);
+    const bool is_voronoi_misconstructed = has_missing_voronoi_vertex || ! is_voronoi_planar;
+    std::unordered_map<Point, Point> vertex_mapping;  // NOTE: Should maybe add a functor to specify specialized hash as 3rd template parameter.
+    Polygons polys_copy = polys; // ATTN!: Don't move inside the scope below, 'segments' and 'vertex_mapping' above both can reference points out of it!
```

Comment:
```suggestion
    Polygons polys_copy; // ATTN!: Don't move inside the scope below, 'segments' and 'vertex_mapping' above both can reference points out of it!
```
Minor performance comment: leave it zero-initialized here, do the copy in the conditional block.